oeq-ts-rest-api/src/AxiosInstance.ts

```diff
@@ -13,20 +14,30 @@ const catchHandler = (error: AxiosError | Error): never => {
   throw repackageError(error);
 };
 
+/**
+ * Executes a HTTP GET for a given path.
+ *
+ * @param path The URL path for the target GET
+ * @param validator A function to perform runtime type checking against the result - typically with typescript-is
+ * @param queryParams The query parameters to send with the GET request
+ * @param transformer A function which returns a copy of the raw data from the GET with any required values transformed - this should NOT mutate the input data (transforms should start on a copy/clone of the input)
+ */
 export const GET = <T>(
   path: string,
   validator?: (data: unknown) => boolean,
```

Comment:
to ensure `validator` is a predicate use
```suggestion
  validator?: (data: unknown) => data is T,
```

Reply:
see: https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates for more information on predicates

Reply:
Hold on though. This is not just meant to be a simple validator or type predicate. This is so that we can parse in a function that under the hood will do runtime validation via typescript-is.

Reply:
Ahhh, okay you're talking about the `typeof` predicate. My bad... Pondering... :thinking:

Reply:
See https://github.com/PenghaiZhang/openEQUELLA/pull/2 for how this would work